packages/memory/src/index.ts

```diff
@@ -549,49 +556,39 @@ export class Memory extends MastraMemory {
 
     const config = this.getMergedThreadConfig(memoryConfig);
 
+    let embeddingTexts: {
+      textForEmbedding: string;
+      message: (typeof messages)[number];
+    }[] = [];
+
+    if (this.vector && config.semanticRecall) {
+      embeddingTexts = updatedMessages.reduce(
+        (acc: { textForEmbedding: string; message: (typeof messages)[number] }[], message) => {
+          const textForEmbedding: string | null = this.prepMessageForEmbedding(message);
+          if (!textForEmbedding) return acc;
+
+          const chunks = this.chunkText(textForEmbedding);
+          if (message.type === 'v2') {
+            if (!message.content.metadata) message.content.metadata = {};
+            message.content.metadata.vectorChunkCount = chunks.length;
+          }
```

Comment:
this will break if v1 messages are passed in, they should be converted to v2 first with message list

Reply:
needs a bit work on top of this but you can do something like: `const updatedMessages = new MessageList().add(messages, 'memory').get.all.v2()`